### PR TITLE
addToCartTest is now functional

### PR DESCRIPTION
**Pull request by SerineBetraoui**
- generated constructor in addToCartTest class
- created retrieveToken method (hashmap, contains find token value gpath) in ProductPage class, accepts productId parameter like selectProduct method, and placed it in the selectProduct method instead of hardcoding the token
- selectProduct method still accepts 2 values, id and quantity

Small changes to the return in getGpathFromXMLBody method in BasePage class

CartPage class added, contains getCartPage method and getProductLineInfo

### CODE REVIEW

File: src/test/java/PageObjects/BasePage.java [hunk 1]
```diff
@@ -29,7 +30,8 @@ public void assertStatusCode(int statusCode) {
 
     public String getGpathFromXmlBody(String gPath) {
         XmlPath xmlPath = new XmlPath(XmlPath.CompatibilityMode.HTML, getResponseBody());
-        return pageElement = xmlPath.get(gPath).toString();
+        pageElement = xmlPath.get(gPath).toString();
```
Comment (Leosaw) [inline src/test/java/PageObjects/BasePage.java:33]: did we need this change I thought It does the same thing?

File: src/test/java/PageObjects/BasePage.java [hunk 2]
```diff
@@ -45,8 +47,17 @@ public Response sendGetRequest(String endpoint) {
         return response;
     }
 
+    public Response sendGetRequest(String path,
+                                   Map<String, Object> queryParams) {
+        response = httpRequest
+                .filter(cookieFilter)
```
Comment (Leosaw) [inline src/test/java/PageObjects/BasePage.java:53]: quite useful to have we can concat a few things here

File: src/test/java/PageObjects/ProductPage.java [hunk 3]
```diff
@@ -1,4 +1,35 @@
...
+        formData.put("group[3]", "13");
+        formData.put("qty", productQuantity);
+        formData.put("add", "1");
+        formData.put("action", "update");
+
+
+        response = httpRequest.filter(cookieFilter)
+                .queryParam("controller", "cart")
+                .formParams(formData)
+                .post("/");
```
Comment (Aayush-Srikanth) [inline src/test/java/PageObjects/ProductPage.java:33]: Could be worth using the sendPostRequest method here for consistency

**Review by Leosaw — APPROVED** [review]

**Review by Aayush-Srikanth — APPROVED** [review]
Nice work!

**Review by jacktegart — APPROVED** [review]

**Pull request merged**